public/app/features/serviceaccounts/ServiceAccountsTable.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { FC, useEffect, useState } from 'react';
+import { AccessControlAction, Role, OrgServiceaccount } from 'app/types';
+import { OrgRolePicker } from '../admin/OrgRolePicker';
+import { Button, ConfirmModal } from '@grafana/ui';
+import { OrgRole } from '@grafana/data';
+import { contextSrv } from 'app/core/core';
+import { fetchBuiltinRoles, fetchRoleOptions, UserRolePicker } from 'app/core/components/RolePicker/UserRolePicker';
+
+export interface Props {
+  serviceaccounts: OrgServiceaccount[];
+  orgId?: number;
+  onRoleChange: (role: OrgRole, serviceaccount: OrgServiceaccount) => void;
+  onRemoveServiceaccount: (serviceaccount: OrgServiceaccount) => void;
+}
+
+const ServiceaccountsTable: FC<Props> = (props) => {
+  const {
+    serviceaccounts: serviceaccounts,
+    orgId,
+    onRoleChange,
+    onRemoveServiceaccount: onRemoveserviceaccount,
+  } = props;
+  const canUpdateRole = contextSrv.hasPermission(AccessControlAction.OrgUsersRoleUpdate);
+  const canRemoveFromOrg = contextSrv.hasPermission(AccessControlAction.OrgUsersRemove);
+  const rolePickerDisabled = !canUpdateRole;
+
+  const [showRemoveModal, setShowRemoveModal] = useState<string | boolean>(false);
+  const [roleOptions, setRoleOptions] = useState<Role[]>([]);
+  const [builtinRoles, setBuiltinRoles] = useState<{ [key: string]: Role[] }>({});
+
+  useEffect(() => {
+    async function fetchOptions() {
+      try {
+        let options = await fetchRoleOptions(orgId);
+        setRoleOptions(options);
+        const builtInRoles = await fetchBuiltinRoles(orgId);
+        setBuiltinRoles(builtInRoles);
+      } catch (e) {
+        console.error('Error loading options');
```

Comment:
Should this situation be notified)somewhere else other than console (e.g. to the user; notification alert or similar)?
Just wondering 🤔

Reply:
I think this is something that we have on the frontend in the users table as well:
https://github.com/grafana/grafana/blob/main/public/app/features/users/UsersTable.tsx#L31

this is strictly taken from there